### PR TITLE
Updates from downstream edsl

### PR DESCRIPTION
**Pull request by j-hui**
Some proposed changes to this library, from working on integration in the downstream edsl repo

- Remove `ssm-debug.h`
- Use `SSM_CRASH` rather than `SSM_RESOURCES_EXHAUSTED`

### CODE REVIEW

**Comment by sedwards-lab** [comment]
Looks fine to me

File: include/ssm.h [hunk 1]
```diff
@@ -68,14 +68,49 @@ See [the detailed documentation](@ref all)
...
+ * Platforms may extend the list of errors using SSM_PLATFORM_ERROR like this:
+ *
+ *      enum {
+ *        SSM_CUSTOM_ERROR1 = SSM_PLATFORM_ERROR,
+ *        // etc.
+ *      };
+ */
+enum ssm_error_t {
+  /** No error, but abruptly halts process without cleanup. */
+  SSM_TERMINATE = 0,
```
Comment (j-hui) [inline include/ssm.h]: Remove this

File: include/ssm.h [hunk 2]
```diff
@@ -68,14 +68,49 @@ See [the detailed documentation](@ref all)
...
+ * exhausted. Not expected to return.
  *
- * Argument passed is a string indicating where the failure occurred.
+ * Argument passed is an ssm_error_t indicating why the failure occurred.
+ * Default behavior is to exit with reason as the exit code, but can be
+ * overridden by defining ssm_crash.
  */
-#define SSM_RESOURCES_EXHAUSTED(string) exit(1)
-#endif
+#define SSM_CRASH(reason) \
```
Comment (j-hui) [inline include/ssm.h]: rename this

File: include/ssm.h [hunk 3]
```diff
@@ -68,14 +68,49 @@ See [the detailed documentation](@ref all)
...
+  /** Reserved for unforeseen, non-user-facing errors. */
+  SSM_INTERNAL_ERROR,
+  /** Tried to insert into full activation record queue. */
+  SSM_EXHAUSTED_ACT_QUEUE,
+  /** Tried to insert into full event queue. */
+  SSM_EXHAUSTED_EVENT_QUEUE,
+  /** Could not allocate more memory. */
+  SSM_EXHAUSTED_MEMORY,
+  /** Tried to exceed available recursion depth. */
+  SSM_EXHAUSTED_DEPTH,
```
Comment (j-hui) [inline include/ssm.h]: rename this to SSM_EXHAUSTED_PRIORITY

**Review by j-hui — COMMENTED** [review]
Some renaming suggestions from Stephen

**Pull request closed without merging**